fix(operator-bundle): use common environment and don't append (#14257)
stackrox/stackrox: operator/bundle_helpers/README.md

For hermetic builds with Konflux, we need to provide the full list of resolved dependencies in `requirements.txt`.
The dependency source files will be prefetched with Cachi2 and made available to the container image build.
Follow the procedure below after any dependencies change for successful builds in Konflux.

## Prepare the fully resolved requirements files for Cachi2

### Prerequisite

> Run the steps inside a container of the same image as the [operator-bundle builder stage](./konflux.bundle.Dockerfile).

Reviewer: The link doesn't work (you can try [here](https://github.com/stackrox/stackrox/blob/tm/operator-bundle-helpers-improvements/operator/bundle_helpers/README.md#prerequisite)) because the file is one level up.

Author: Done in https://github.com/stackrox/stackrox/pull/14257/commits/817a312c46b790f483f0e147f223493ec2872f67

```bash

> cd $(git rev-parse --show-toplevel)/operator/bundle_helpers
> docker run -it -v $(pwd):/src --entrypoint /bin/bash -w /src registry.access.redhat.com/ubi9/python-39:latest

Reviewer: This is not what I suggested. My suggestion was that while you're anywhere in the repo, you can run the command and it does not move your location.
```suggestion
docker run -it -v "$(git rev-parse --show-toplevel)/operator/bundle_helpers:/src" --entrypoint /bin/bash -w /src registry.access.redhat.com/ubi9/python-39:latest
```

Author: done in https://github.com/stackrox/stackrox/pull/14257/commits/817a312c46b790f483f0e147f223493ec2872f67

For hermetic builds with Konflux, we need to provide the full list of resolved dependencies in `requirements.txt`.
The dependency source files will be prefetched with Cachi2 and made available to the container image build.
Because GHA/upstream is running on a different Python version, it uses `requirements-gha.txt` to mange its own dependencies.

> Follow the procedure below after any dependencies change for successful builds in Konflux and mirror those changes into the `requirements-gha.txt`.

Reviewer: Nitpick

> Follow the procedure below ... and mirror those changes into the `requirements-gha.txt`.

This can be read like there's 1:1 copy in `requirements-gha.txt` or there's some tooling/process, whereas no tooling exists and in practice we're expected to creatively manually maintain `requirements-gha.txt`. Could you please adjust the sentence to prevent misleading. I'd even suggest to remove the mention of `requirements-gha.txt` from here and add a small section about dealing with that file.

## Prepare the fully resolved requirements files for Cachi2

### Prerequisite

> Run the steps inside a container of the same image as the operator-bundle builder stage.

Reviewer: Here I'd suggest to hint readers at `konflux.bundle.Dockerfile`, that's where they can find the image of the operator-bundle builder stage.

```bash

> cd operator/bundle_helpers # if not already in this directory
> docker run -it -v $(pwd):/src --entrypoint /bin/sh registry.access.redhat.com/ubi9/python-39:latest
> # inside the container
> cd /src

Reviewer: 1. `operator/bundle_helpers` can be handled simpler: figure the repo root and go from there.
2. Let's just use `bash` because `sh` is an alias of `bash`.
3. You don't need to `cd /src` when you can set working directory right away with `-w`.

```suggestion
docker run -it -v "$(git rev-parse --show-toplevel)/operator/bundle_helpers:/src" -w /src --entrypoint /bin/bash registry.access.redhat.com/ubi9/python-39:latest
# run the following commands inside the container
```

Author: done in https://github.com/stackrox/stackrox/pull/14257/commits/dbeb718219a34cb7575150de80d8fa323b21ebaf

```bash
./pip_find_builddeps.py requirements.txt \
  -o requirements-build.in

pip-compile requirements-build.in --allow-unsafe --generate-hashes
```

> 

Reviewer: A reminder to commit changes will be handy.

Author: done in https://github.com/stackrox/stackrox/pull/14257/commits/dbeb718219a34cb7575150de80d8fa323b21ebaf

pip-compile requirements-build.in --allow-unsafe --generate-hashes
```

4. Exit the container and commit the changes.

For more information, consult the [Cachi2 docs](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#building-from-source).

### What does each requirements file do?

* `requirements.in`: List of project dependencies.

> * `requirements-gha.txt`: Temporary list of project dependencies as required by the build process on GHA and locally. This workaround exists due to a different Python version. Any changes in this or the `requirements.in` file must be synced manually to the other. This will be deleted after ROX-26860.

Reviewer: ```suggestion
* `requirements-gha.txt`: List of project dependencies as required by the build process on GHA and locally. This file exists as a workaround due to a different Python version in this context. Any changes in this or the `requirements.in` file should be synced manually to `requirements-gha.txt`. This file will be deleted after ROX-26860.
```

Author: done in https://github.com/stackrox/stackrox/pull/14257/commits/817a312c46b790f483f0e147f223493ec2872f67
